delfin/drivers/fujitsu/eternus/eternus_stor.py

```diff
@@ -352,3 +391,548 @@ def parse_alert(context, alert):
+            if len(host_fc) < consts.HOST_TOTAL:
+                continue
+            fc_name = host_fc[consts.HOST_NAME_COUNT]
+            os = host_fc[consts.HOST_TYPE_COUNT].lower()
+            state = host_status.get(fc_name)
+            host_d = {
+                "name": fc_name,
+                "storage_id": self.storage_id,
+                "native_storage_host_id": fc_name,
+                "os_type": consts.HOST_OS_TYPES_MAP.get(
+                    os, constants.HostOSTypes.UNKNOWN),
+                "status": constants.HostStatus.NORMAL if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.HostStatus.OFFLINE
+            }
+            host_list.append(host_d)
+
+        host_iscsi_list = self.get_iscsi_host()
+        for host_iscsi in (host_iscsi_list or []):
+            iscsi_name = host_iscsi.get('name')
+            state = host_status.get(iscsi_name)
+            os = host_iscsi.get('os')
+            os = os.lower() if os else None
+            host_d = {
+                "name": iscsi_name,
+                "storage_id": self.storage_id,
+                "native_storage_host_id": iscsi_name,
+                "os_type": consts.HOST_OS_TYPES_MAP.get(
+                    os, constants.HostOSTypes.UNKNOWN),
+                "status": constants.HostStatus.NORMAL if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.HostStatus.OFFLINE,
+                'ip_address': host_iscsi.get('address')
+            }
+            host_list.append(host_d)
+
+        host_sas_list = self.get_data(consts.GET_HOST_SAS_ADDRESSES,
+                                      consts.HOST_SAS_FIVE)
+        for host_sas in (host_sas_list or []):
+            if len(host_sas) < consts.HOST_SAS_FIVE:
+                continue
+            sas_name = host_sas[consts.HOST_SAS_ONE]
+            sas_os = host_sas[consts.HOST_SAS_FOUR].lower()
+            state = host_status.get(sas_name)
+            host_d = {
+                "name": sas_name,
+                "storage_id": self.storage_id,
+                "native_storage_host_id": sas_name,
+                "os_type": consts.HOST_OS_TYPES_MAP.get(
+                    sas_os, constants.HostOSTypes.UNKNOWN),
+                "status": constants.HostStatus.NORMAL if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.HostStatus.OFFLINE
+            }
+            host_list.append(host_d)
+        return host_list
+
+    def get_data(self, command, length_count):
+        host_list = []
+        host_str = self.cli_handler.exec_command(command)
+        block = True
+        length_list = []
+        if host_str:
+            host_arr = host_str.strip().replace('\r', '').split('\n')
+            for host_row_str in (host_arr or []):
```

Comment:
The return value of the split() method is a list, and no additional judgment of or [] is required.
![image](https://user-images.githubusercontent.com/59107077/153789598-1b92c62c-9393-4aed-9a9d-45ef75e35ce3.png)

Reply:
Has been modified